src/fastoad/openmdao/utils.py

```diff
@@ -92,11 +92,20 @@ def get_problem_after_setup(problem: om.Problem) -> om.Problem:
     having the problem being actually setup.
 
     :param problem:
-    :return: the problem itself it setup() has already been run, or a copy of the provided problem after setup()
-             has been run
+    :return: the problem itself it setup() has already been run, or a copy of the provided problem
+             after setup() has been run
     """
 
-    if problem._setup_status == 0:
+    if version.parse(openmdao.__version__) < version.parse("3.3"):
+        problem_is_setup = problem._setup_status != 0
+    else:
+        from openmdao.core.constants import _SetupStatus
+
+        problem_is_setup = (
+            problem._metadata and problem._metadata["setup_status"] >= _SetupStatus.POST_SETUP
```

Comment:
J'ai un doute sur le besoin de mettre `problem._metadata` car si il est vide `problem._metadata["setup_status"]` renverra une erreur quand même, non ?

Reply:
Justement non, c'est la façon standard d'éviter une erreur, car Python est économe: dans un `and`, Python évalue successivement les assertions et s'arrête d'évaluer dès que une est fausse (pareil dans un `or`, mais il s'arrête dès que une est vraie)

Pour le vérifier:
```python
l = {} # works also with `l = None`
if l and l["foo"]:
    print("Unexpected!!!")
else:
    print("Expected.")
```

Reply:
J'avais jamais fait gaffe, sympa !